### DISCUSSION TITLE
doris使用问题汇总，看到的大佬辛苦给一些建议

### DISCUSSION THREAD

**Discussion by 980150976** [apache/doris#9616] [A - General / Q&A]
1 集群的节点数量：3 FE(1 leader,2 follower)  + BE建议多少节点？，节点的配置有什么建议（多少核，多少内存，磁盘配置等【考虑冷热数据】）
2 Apache Doris (incubating) 1.0 Release 版本中，Doris Manger 可视化监控运维平台什么时候可以发布？发布的版本是否已经支持array数据类型【看功能已经merge】。
3 如果每个维度（100列，100亿行）都加bitmap或者bloom filter索引（假设合理使用索引），具体的空间占用会有多大【比较不加索引情况下】？ 这两种索引加的过多会影响写入性能吗？
4 doris 对数据更新能力支持怎么样，内部基于compaction合并，更新时效性可以在表参数中控制吗？
5 doris String类型底层用的压缩算法是什么，压缩比预计能到多少？；在字段字符大小不确定的情况下使用String类型有坑吗

**Comment by dataalive** [comment]
尝试回答部分问题

1. BE建议3-4个起（数据3副本的考虑），实际上FE和BE也可以混布，当然前提是资源需要分配好。建议最小集开始，按照业务压力逐步扩容即可，因为Doris支持水平扩展比较好。节点配置见过主流的是16核64G居多，具体CPU、内存这些都跟业务实际需求有关系。冷热数据的话
可以考虑配置SSD + HDD（如果是云上的话就更方便了），表建立的时候可以配置cooldown的配置能自动下沉。

2. Doris Manager目前支持节点的可视化部署，再配合Grafna等来做监控。

5. 默认为lz4，ssb的数据的压缩比为1:4到1:5，如果日志类的数据压缩比还可以更高。字符大小不确定但是有个最大长度把？

**Reply by LOVEGISER** [reply]
**1.常用数据类型如下:**

BOOLEAN（1字节）
        范围：{0,1}
    TINYINT（1字节）
        范围：-2^7 + 1 ~ 2^7 - 1
    SMALLINT（2字节）
        范围：-2^15 + 1 ~ 2^15 - 1
    INT（4字节）
        范围：-2^31 + 1 ~ 2^31 - 1
    BIGINT（8字节）
        范围：-2^63 + 1 ~ 2^63 - 1
    LARGEINT（16字节）
        范围：-2^127 + 1 ~ 2^127 - 1
    FLOAT（4字节）
        支持科学计数法
    DOUBLE（8字节）
        支持科学计数法
    DECIMAL[(precision, scale)] (16字节)
        保证精度的小数类型。默认是 DECIMAL(10, 0)
        precision: 1 ~ 27
        scale: 0 ~ 9
        其中整数部分为 1 ~ 18
        不支持科学计数法
    DATE（3字节）
        范围：0000-01-01 ~ 9999-12-31
    DATETIME（8字节）
        范围：0000-01-01 00:00:00 ~ 9999-12-31 23:59:59
    CHAR[(length)]
        定长字符串。长度范围：1 ~ 255。默认为1
    VARCHAR[(length)]
        变长字符串。长度范围：1 ~ 65533
    HLL (1~16385个字节)
        hll列类型，不需要指定长度和默认值、长度根据数据的聚合
        程度系统内控制，并且HLL列只能通过配套的hll_union_agg、Hll_cardinality、hll_hash进行查询或使用
    BITMAP
        bitmap列类型，不需要指定长度和默认值。表示整型的集合，元素最大支持到2^64 - 1

**2.定义列时，可参照如下建议：**

Key 列必须在所有 Value 列之前。
尽量选择整型类型。因为整型类型的计算和查找比较效率远高于字符串。
对于不同长度的整型类型的选择原则，遵循 够用即可。
对于 VARCHAR 和 STRING 类型的长度，遵循 够用即可。
所有列的总字节长度（包括 Key 和 Value）不能超过 100KB。

**Comment by yangunang** [comment]
我这边使用版本是 1.0.0-rc02-9928900 ，操作系统是centOS Linux release 7.6.1810，现在报Unsupported data type: ARRAY<int(11)>，是不支持array数据类型吗？

**Reply by dataalive** [reply]
array这个在你的版本中还没支持。大概率会在1.2发布
